typing/includecore.ml

```diff
@@ -33,18 +79,24 @@ let value_descriptions ~loc env name
     loc
     vd1.val_attributes vd2.val_attributes
     name;
-  if Ctype.moregeneral env true vd1.val_type vd2.val_type then begin
-    match (vd1.val_kind, vd2.val_kind) with
-        (Val_prim p1, Val_prim p2) ->
-          if p1 = p2 then Tcoerce_none else raise Dont_match
+  match Ctype.moregeneral env true vd1.val_type vd2.val_type with
+  | exception Ctype.Moregen trace -> raise (Dont_match (Type (env, trace)))
+  | () -> begin
```

Comment:
We provide both `Ctype.moregeneral`, which raises, and `Ctype.is_moregeneral`, which returns a boolean, so that either raising or not raising can be accomplished as desired.

Reply:
I think it is good to have these functions match the behaviour of `unify`. We can decide later if we want to move all these functions over to using a result type.